modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql

```diff
@@ -9163,6 +9135,197 @@ type GmosNorthLongSlit {
+
+  "Interleaved mode cycles through all filters repeatedly."
+  interleaved: GmosInterleavedImagingVariantInput
+
+  "PreImaging mode is used for MOS mask creation."
+  preImaging:  GmosPreImagingVariantInput
+}
+
+"""
+When doing "grouped" filter imaging, datasets associated with a particular
+filter are collected consecutively before moving on to other filters (if any).
+Sky datasets may be collected before and after each group of object datasets.
+"""
+type GmosGroupedImagingVariant {
+
+  """
+  Whether the filters should appear in the sequence in increasing or decreasing
+  order by their wavelength.
+  """
+  order: WavelengthOrder!
+
+  """
+  Offset generator for the science object datasets. The same offset sequence is
+  created for each filter using the specified generator.
+  """
+  offsets: OffsetGenerator
+
+  """
+  Number of sky positions to collect before and after object datasets. For
+  example, if set to 2 there will be two sky positions before a group of object
+  exposures and two more after using the same filter as the object datasets.
+  """
+  skyCount: NonNegInt!
+
+  """
+  Offset generator to use for the sky datasets.
+  """
+  skyOffsets: OffsetGenerator
+}
+
+"""
+Input used for specifying GMOS grouped filter imaging.
+"""
+input GmosGroupedImagingVariantInput {
+
+  """
+  Whether the filters should appear in the sequence in increasing or decreasing
+  order by their wavelength.  Defaults to `INCREASING` on create, absent (i.e.,
+  not modified) on update.
+  """
+  order: WavelengthOrder
+
+  """
+  Offset generator for the science object datasets. The same offset sequence is
+  created for each filter using the specified generator.  If not specified, no
+  offsets will be used.
+  """
+  offsets: OffsetGeneratorInput
+
+  """
+  Number of sky positions to collect before and after object datasets. For
+  example, if set to 2 there will be two sky positions before a group of object
+  exposures and two more after using the same filter as the object datasets.
+  Defaults to 0 on creation, absent (i.e., not modified) on update.
+  """
+  skyCount: NonNegInt
+
+  """
+  Offset generator to use for sky positions.  If not specified, no offsets will
+  be used.
+  """
+  skyOffsets: OffsetGeneratorInput
+
+}
+
+"""
+When doing "interleaved" filter imaging, the sequence repeatedly alternates
+through the set of filters in use.
+"""
+type GmosInterleavedImagingVariant {
+
+  """
+  Offset generator for the science object datasets. The offset pattern is
+  applied to the sequence of science datasets as a whole.
+  """
+  offsets: OffsetGenerator
+
+  """
+  Number of sky positions to collect, per filter, before and after a series of
+  object datasets.  For example, if set to 2 and 2 filters are in use, there
+  would be 4 sky positions before (2 per filter) and 4 after (2 per filter).
+  """
+  skyCount: NonNegInt!
+
+  """
+  Offset generator to use for the sky datasets.
+  """
+  skyOffsets: OffsetGenerator
```

Comment:
`skyCount` and `skyOffsets` are related of course.  It would probably be best to add additional validation. For example, when `skyOffsets` are defined on creation then `skyCount` should be at least 1 (otherwise no sky positions would be generated regardless).